src/main/java/org/spongepowered/api/service/event/EventManager.java

```diff
@@ -34,13 +35,23 @@
     /**
      * Registers an object to receive {@link Event}s.
      *
-     * @param plugin A plugin instance
+     * @param plugin The plugin instance
      * @param obj The object
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin instance
      */
     void register(Object plugin, Object obj);
```

Comment:
I guess we should mention here that this method searches for annotations, or how the input is used at all.
For example what happens if we pass in an EventHandler? (For example if we forgot the class param)

Maybe we should rename the method to make it more clear.
`registerAnnotatedHandlers(...)`

We also should return the actually registered listeners from this method
`Collection/Set<EventHandler> registerAnnotatedHandlers(Object plugin, Object obj)`;